experimental/yarpl/include/yarpl/Flowable.h

```diff
@@ -2,48 +2,87 @@
 
 #pragma once
 
-#include <functional>
 #include <memory>
-#include "yarpl/Flowable_Subscriber.h"
+#include <type_traits>
+#include <utility>
+#include "yarpl/flowable/sources/Flowable_RangeSubscription.h"
+
+#include "reactivestreams/ReactiveStreams.h"
+#include "yarpl/utils/type_traits.h"
 
 namespace yarpl {
 namespace flowable {
 
+// using reactivestream_yarpl to not conflict with the other reactivestreams
+
 template <typename T>
-class Flowable {
+class Flowable : public reactivestreams_yarpl::Publisher<T>,
+                 public std::enable_shared_from_this<Flowable<T>> {
+  using Subscriber = reactivestreams_yarpl::Subscriber<T>;
+  using Subscription = reactivestreams_yarpl::Subscription;
+
  public:
-  explicit Flowable(
-      std::function<void(std::unique_ptr<Subscriber<T>>)> onSubscribe);
+  template <
+      typename F,
+      typename = typename std::enable_if<
+          std::is_callable<F(std::unique_ptr<Subscriber>), void>::value>::type>
+  static std::unique_ptr<Flowable> create(F&& onSubscribeFunc) {
```

Comment:
I apologize, I'm kind of slow ... read through that, and re-read Item 25 about perfect forwarding in "Effective Modern C++", but am not understanding what I'm doing wrong here. 

What needs to change?

Here is the current method:

```cpp
template <
    typename T,
    typename F,
    typename = typename std::enable_if<std::is_callable<
        F(std::unique_ptr<reactivestreams_yarpl::Subscriber<T>>),
        void>::value>::type>
static auto unsafeCreateUniqueFlowable(F&& onSubscribeFunc) {
  return std::make_unique<UniqueFlowable<T, F>>(
      std::forward<F>(onSubscribeFunc));
}
```